tests/unit/core/keypairs/test_main.py

```diff
@@ -49,6 +49,19 @@ def test_derive_keypair_ed25519_validator(self):
         with self.assertRaises(XRPLKeypairsException):
             keypairs.derive_keypair("sEdSKaCy2JT7JaM7v95H9SxkhP9wS2r", validator=True)
 
+    def test_derive_keypair_ed25519_different_prefix(self):
+        public, private = keypairs.derive_keypair(
+            "ssB9S5Mca2hGZ73xNs4gruS1GY7fB", algorithm=CryptoAlgorithm.ED25519
+        )
+        self.assertEqual(
+            public,
+            "ED6BBFC23A490D021B87D25563C15DA953A7F0F1A493DAA3767FB27F82E2F80C3D",
+        )
+        self.assertEqual(
+            private,
+            "ED644E705250E4D736875E85DD3E5FBABA4E12E004549202010228E17D3D574576",
+        )
+
```

Comment:
Can you add a test case for when a normal ED25519 key is given with the CryptoAlgorithm set to SECP256K1? (That way the exception is thrown for an invalid decoding)